test/acceptance/formatters/kakoune-test.js

```diff
@@ -0,0 +1,217 @@
+    await project.setConfig({
+      rules: {
+        'require-button-type': true,
+      },
+    });
+
+    await project.write({
+      app: {
+        components: {
+          'click-me-button.hbs': '<button>Click me!</button>',
+        },
+      },
+    });
+
+    let result = await runBin('.', '--format', 'kakoune');
+
+    expect(result.exitCode).toEqual(1);
+    expect(result.stdout).toMatchInlineSnapshot(`
+      "app/components/click-me-button.hbs:1:1: Error: All \`<button>\` elements should have a valid \`type\` attribute (require-button-type) 
+
+      1 problem"
+    `);
+
+    expect(result.stderr).toBeFalsy();
+  });
+
+  it('should output to a file using --output-file option using default filename', async () => {
+    await project.setConfig({
+      rules: {
+        'no-bare-strings': true,
+        'no-html-comments': 'warn',
+      },
+    });
+    await project.write({
+      app: {
+        templates: {
+          'application.hbs':
+            '<h2>Here too!!</h2><div>Bare strings are bad...</div><!-- bad html comment! -->',
+        },
+      },
+    });
+
+    let result = await runBin('.', '--format', 'kakoune', '--output-file');
+
+    expect(result.exitCode).toEqual(1);
+    expect(getOutputFileContents(result.stdout)).toMatchInlineSnapshot(`
+      "app/templates/application.hbs:1:4: Error: Non-translated string used (no-bare-strings) 
+      app/templates/application.hbs:1:24: Error: Non-translated string used (no-bare-strings) 
+      app/templates/application.hbs:1:53: Warning: HTML comment detected (no-html-comments) 
+
+      3 problems"
+    `);
+    expect(result.stderr).toBeFalsy();
+  });
+
+  it('should output to a file using --output-file option using custom filename', async () => {
+    await project.setConfig({
+      rules: {
+        'no-bare-strings': true,
+        'no-html-comments': 'warn',
+      },
+    });
+    await project.write({
+      app: {
+        templates: {
+          'application.hbs':
+            '<h2>Here too!!</h2><div>Bare strings are bad...</div><!-- bad html comment! -->',
+        },
+      },
+    });
+
+    let result = await runBin(
+      '.',
+      '--format',
+      'kakoune',
+      '--output-file',
+      'kakoune-output.kakoune'
+    );
+
+    expect(result.exitCode).toEqual(1);
+    expect(result.stdout).toMatch(/.*kakoune-output\.kakoune/);
+    expect(getOutputFileContents(result.stdout)).toMatchInlineSnapshot(`
+      "app/templates/application.hbs:1:4: Error: Non-translated string used (no-bare-strings) 
+      app/templates/application.hbs:1:24: Error: Non-translated string used (no-bare-strings) 
+      app/templates/application.hbs:1:53: Warning: HTML comment detected (no-html-comments) 
+
+      3 problems"
+    `);
+    expect(result.stderr).toBeFalsy();
+  });
+
+  describe('with --quiet option', function () {
+    it('should print valid JSON string with errors, omitting warnings', async function () {
```

Comment:
1. Where is the JSON string?
2. Which rule in this test had warning violations that were silenced?

Reply:
The test was originally from the json test. Is fixed now.